pkg/states/state_manager.go

```diff
@@ -0,0 +1,83 @@
+package states
+
+import "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+func Active(userSignup *v1alpha1.UserSignup) bool {
+	return Approved(userSignup) &&
+		!VerificationRequired(userSignup) &&
+		!Deactivated(userSignup)
+}
+
+func Approved(userSignup *v1alpha1.UserSignup) bool {
+	return contains(userSignup.Spec.States, v1alpha1.UserSignupStateApproved)
+}
+
+func SetApproved(userSignup *v1alpha1.UserSignup, val bool) {
+	if val && !contains(userSignup.Spec.States, v1alpha1.UserSignupStateApproved) {
+		userSignup.Spec.States = append(userSignup.Spec.States, v1alpha1.UserSignupStateApproved)
+	}
+
+	if !val && contains(userSignup.Spec.States, v1alpha1.UserSignupStateApproved) {
+		userSignup.Spec.States = remove(userSignup.Spec.States, v1alpha1.UserSignupStateApproved)
+	}
+}
+
+func VerificationRequired(userSignup *v1alpha1.UserSignup) bool {
+	return contains(userSignup.Spec.States, v1alpha1.UserSignupStateVerificationRequired)
+}
+
+func SetVerificationRequired(userSignup *v1alpha1.UserSignup, val bool) {
+	if val && !contains(userSignup.Spec.States, v1alpha1.UserSignupStateVerificationRequired) {
+		userSignup.Spec.States = append(userSignup.Spec.States, v1alpha1.UserSignupStateVerificationRequired)
+	}
+
+	if !val && contains(userSignup.Spec.States, v1alpha1.UserSignupStateVerificationRequired) {
+		userSignup.Spec.States = remove(userSignup.Spec.States, v1alpha1.UserSignupStateVerificationRequired)
+	}
+}
+
+func Deactivating(userSignup *v1alpha1.UserSignup) bool {
+	return contains(userSignup.Spec.States, v1alpha1.UserSignupStateDeactivating)
+}
+
+func SetDeactivating(userSignup *v1alpha1.UserSignup, val bool) {
+	if val && !contains(userSignup.Spec.States, v1alpha1.UserSignupStateDeactivating) {
+		userSignup.Spec.States = append(userSignup.Spec.States, v1alpha1.UserSignupStateDeactivating)
+	}
+
+	if !val && contains(userSignup.Spec.States, v1alpha1.UserSignupStateDeactivating) {
+		userSignup.Spec.States = remove(userSignup.Spec.States, v1alpha1.UserSignupStateDeactivating)
+	}
```

Comment:
I guess we can add a generic function like:
```
func setState(usersignup, val, stateType)
```
And reuse it.

Reply:
That makes a lot of sense. :)